### DISCUSSION TITLE
Connect two data centers using ceph

### DISCUSSION THREAD

**Discussion by gabriel-sandoli** [rook/rook#13331] [Q&A] [answered]
Hi guys, i have 2 data centers with 4 machines each, and have successfully installed ceph on each of them. But now i have to connect the two data centers to share files between them, can i do this using another cluster running inside one of my kubernetes clusters? In other words, can i have one ceph cluster that connect to a machine that is not on my kubernetes cluster? If not, can i install ceph via cephadm without breaking my existing cluster?

**Comment by travisn** [comment]
From where do you want to consume the common storage? Do you want the two K8s clusters to share data from the Ceph cluster? In that case, the common scenario is to pick one cluster where Ceph is running, then the other cluster can access that Ceph with an [External cluster configuration](https://rook.io/docs/rook/latest/CRDs/Cluster/external-cluster/). This requires some network connection between the cluster, and typically Ceph will be configured with host networking. cephadm can also be used to configure the external cluster, but if you already have Rook configuring ceph, not sure why you would need a cephadm cluster.

**Reply by gabriel-sandoli** [reply]
First of all, thanks for the help! What you describe does not quite match what i want. Both K8s cluster needs to ingest common input data, but each cluster will have separate outputs. So basically cluster A will consume the common input data and do it's thing, generating some outputs files that are shared only in cluster A, and cluster B will also consume the common input data, but i'll do it's own thing, generating other outputs files that are shared only in cluster B. I already setup cluster A and B Ceph Cluster, but now i need to create a way to share the common input files, i think it's more clear what i need now.

**Reply by travisn** [reply, marked as answer]
What you need is three separate pools:
1. The common pool, shared by both clusters
2. The output pool for cluster A
3. The output pool for cluster B

Each pool will have a diifferent storage class, and you would provision volumes from those storage classes.

You could choose to have all these pools in a single cluster, or two of them in one of the clusters if you want to keep the ouptut pools in different clusters. But you don't need a third ceph cluster.

**Reply by gabriel-sandoli** [reply]
Hi! You answer still does not answear my original question, but gave me an idea to change how to do things here, and thinking about it, it will be easier, so thanks for the help! Closing the disscussion.

**Comment by VisionWaves-config** [comment]
Hi @gabriel-sandoli  Did you found solution of it? In my case also I have two different k8s cluster with two saparate ceph clusetr and I need to replicate data and k8s should point saparate PV / PVC as per different SITE/DC . How the same data can be claimed on tow different statefulset accross DC?
